### PR TITLE
Employ a `NamedTuple` for TDS parameters

### PR DESCRIPTION
**Pull request by portante**
Like the Tool Meister module, employ a `NamedTuple` for the Tool Data Sink parameters.

Layered on PR #2755.

### CODE REVIEW

File: lib/pbench/agent/tool_data_sink.py [hunk 1]
```diff
@@ -739,28 +750,23 @@ class ToolDataSink(Bottle):
...
+            tdsp = ToolDataSinkParams(
+                benchmark_run_dir=_benchmark_run_dir,
+                bind_hostname=params["bind_hostname"],
+                port=params["port"],
+                channel_prefix=params["channel_prefix"],
+                tool_group=params["group"],
+                tool_metadata=ToolMetadata.tool_md_from_dict(params["tool_metadata"]),
+                tool_trigger=params["tool_trigger"],
+                tools=params["tools"],
+            )
```
Comment (webbnh) [inline lib/pbench/agent/tool_data_sink.py:765]: I would get rid of the local variables:
```
            return ToolDataSinkParams(
                benchmark_run_dir=BenchmarkRunDir(
                    params["benchmark_run_dir"], pbench_run
                ),
                bind_hostname=params["bind_hostname"],
                port=params["port"],
                channel_prefix=params["channel_prefix"],
                tool_group=params["group"],
                tool_metadata=ToolMetadata.tool_md_from_dict(params["tool_metadata"]),
                tool_trigger=params["tool_trigger"],
                tools=params["tools"],
            )

```

**Review by webbnh — DISMISSED** [review]
Easy-peasy.

File: lib/pbench/agent/tool_data_sink.py (hunk 1 above)
Comment (portante) [reply lib/pbench/agent/tool_data_sink.py:765]: So this change came about by breaking a larger commit that was going to land in PR #2743 into smaller ones.  What is coming is a [change](https://github.com/distributed-system-analysis/pbench/pull/2743/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-f186c241cb95e4c5972607dfea8d6669f4cb48c20a0def4276f58ad08808214dR771) to use that variable.

**Review by webbnh — APPROVED** [review]
👍

**Review by dbutenhof — APPROVED** [review]

**Review by npalaska — APPROVED** [review]

**Pull request merged**